src/languageservice/parser/yamlParser07.ts

```diff
@@ -33,7 +33,8 @@ export function parse(text: string, parserOptions: ParserOptions = defaultOption
   };
   const composer = new Composer(options);
   const lineCounter = new LineCounter();
-  const parser = new Parser(lineCounter.addNewLine);
+  const regx = new RegExp('[\\n]+[\\s]*$', 'g');
```

Comment:
I don't really like this rexexp, could we achieve same behaviour without the regexp.

You apply that reg exp to file content, files can be big and it not very efficient to apply such regexps on each re parse.

Reply:
Ya. I have changed the logic. Instead of using RegExp I have used Lexer from yaml for getting tokens of the text.